07-regression1.Rmd

```diff
@@ -510,7 +537,14 @@ plot_pairs <- sacramento %>%
 plot_pairs
 ```
 
-From this we can see that generally, as both house size and number of bedrooms increase, so does price. Does adding the number of bedrooms to our model improve our ability to predict house price? To answer that question, we will have to come up with the test error for a K-NN regression model using house size and number of bedrooms, and then we can compare it to the test error for the model we previously came up with that only used house size to see if it is smaller (decreased test error indicates increased prediction quality). Let's do that now!
+Figure \@ref(fig:07-ggpairs) shows that generally, as both house size and
+number of bedrooms increase, so does price. Does adding the number of bedrooms
+to our model improve our ability to predict house price? To answer that
+question, we will have to estimate the test error for a K-NN regression
+model using house size and number of bedrooms, and then we can compare it to
+the test error for the model we previously came up with that only used house
+size to see if it is smaller (decreased test error indicates increased
```

Comment:
This is breaking the golden rule I think, and goes against what we did in the previous chapter for best subset selection. Here we really should compare the cross-validated error for these two models, not the test error...

Reply:
We can still compare the test error, I guess..., but we should discuss this is done from the context of teaching. And that in practice, we do not look at the test error until we have one final model standing...

Reply:
I think I've made an edit that addresses both the desiderata to (1) show students how to predict on the test set, but also (2) not confuse them into thinking you compare models using the test data.